### ISSUE TITLE
High Cpu usage on jetbrains Ides

### ISSUE THREAD

**Issue by ouedyan** [denofevil/import-cost#49]
Please fix high cpu usage on Jetbrains Ides when your plugin is enabled. For details please see: 
https://youtrack.jetbrains.com/issue/WEB-58351/WebStorm-EAP-high-CPU-usage#focus=Comments-27-6724186.0-0

**Comment by JuHwon** [comment]
i also noticed high cpu usage. fans are spinning high and all cores close to 100% when webstorm is opened. disabled plugin and everything was fine.

**Comment by patric-eberle** [comment]
Same here. I also have increased energy consumption on a `PhpStorm > Node` progress when checking on macOS Activity monitor. As soon as I disabled this plugin, the values returned to normal. This issue only popped up with PhpStorm 2022.3 and was not an issue with PhpStorm 2022.2.

**Comment by Webbrother** [comment]
+1 

It is **REALLY** annoying.

Macbook Pro 2019, 2,4 GHz 8‑core Intel Core i9, 32Gb.
OS: MacOS Ventura 13.2.1 
Webstorm: 2022.3.3 Build #WS-223.8836.27, built on March 1, 2023

It loads CPU on 100%!
The notebook becomes hot! If the webstorm is launched, the battery lasts for **one hour**.

**Comment by na-ji** [comment]
> I will try to reduce a number of thread for the calculation, but unfortunatelly we don't have much control from our side there. Related issue on the library side [wix/import-cost#272](https://github.com/wix/import-cost/issues/272)

This is not an issue with a number of threads, but because the plugin is creating an infinite number of processes at an incredible pace. All those processes are short-lived, less than an instant. In a few seconds, more than 5k processes were created on my machine. It takes less than 2 minutes for the PID of the processes created to be reset from 99999 to 0. 
I could attach a video of the phenomenon seen on the activity monitor, but I doubt it would help on debugging what's happening.

**Comment by anstarovoyt** [comment]
@na-ji thank you! Can you share the project which I can use for testing. Or it works this way for all possible project for you?

**Comment by na-ji** [comment]
@anstarovoyt No, this doesn't happen for all possible project. It's hard to tell, but I noticed at first it happens really fast on monorepos, but it's not limited to those monorepos. After testing a few projects, my theory is that it happens because of typescript import aliases. I confirmed that the issue is happening on this example for instance: https://github.com/vercel/next.js/tree/canary/examples/with-absolute-imports
On this example, it's not hitting the CPU as hard as on a real project, but I still see a noticeable increase and the same node processes that are created really fast.
